### PR TITLE
Fix empty namespace name in WithNamespace (#617)

### LINKED ISSUE #617: Empty namespace name causes unexpected behavior
This code:

```c#
class Fabric : ProjectFabric
{
    public override void AmendProject(IProjectAmender amender)
    {
        amender.AddAspect<IntroduceTypeAspect>();
    }
}

class IntroduceTypeAspect : CompilationAspect
{
    public override void BuildAspect(IAspectBuilder<ICompilation> builder)
    {
        builder.Advice.WithNamespace(builder.Target.GlobalNamespace, "").IntroduceClass("Foo");
    }
}
```

Generates the invalid code:

```c#


namespace {
    class Foo
    {
    }
}
```

Empty name namespace should either be considered the same  as the parent namespace (the global namespace in this case), or attempting to use it should throw an exception.

### PR DESCRIPTION
**Pull request by PostSharpAgent**
## Summary

Fixes #617 — `WithNamespace(globalNamespace, "")` generates invalid C# code with an empty namespace block.

When an empty string is passed as a namespace name to `WithNamespace` or `WithChildNamespace`, the generated code contained `namespace { }` (invalid syntax) and references like `global::.Foo` (invalid leading dot).

**Fix:** In `AdviceFactory.WithNamespace`, when the name is empty or null, return an adviser for the target namespace directly instead of creating a new namespace with an empty name. This makes `WithNamespace(x, "")` equivalent to `WithNamespace(x)`, as expected.

## Changes

- `Metalama.Framework.Engine/Advising/AdviceFactory.cs` — Added early return for empty name in `WithNamespace`; added `using (this.WithNonUserCode())` to `WithDeclaration` per review feedback
- Two new regression tests in `Tests/Aspects/Introductions/Namespaces/`:
  - `EmptyNameWithNamespace` — `WithChildNamespace("")` on the global namespace
  - `EmptyNameWithNamespaceOnCompilation` — `builder.WithNamespace("")` on `CompilationAspect`

## Progress

- [x] Reproduce bug with regression tests
- [x] Implement fix
- [x] Run all tests in the solution (2044 AspectTests pass)
- [x] Build with `Build.ps1 build` (zero warnings)
- [x] Run `Build.ps1 test` (all tests pass)
- [x] Address review feedback (add `WithNonUserCode()` to `WithDeclaration`)
- [x] Finalize PR

— Claude for @gfraiteur

### CODE REVIEW

**Review by gfraiteur — APPROVED** [review]

File: Metalama.Framework/src/Metalama.Framework.Engine/Advising/AdviceFactory.cs [hunk 1]
```diff
@@ -1898,6 +1898,12 @@ public IAdviser<INamespace> WithNamespace(
             throw new InvalidOperationException();
         }
 
+        // An empty name resolves to the target namespace itself.
+        if ( string.IsNullOrEmpty( name ) )
```
[the remark this answers is not in the document]
Comment (gfraiteur) [reply Metalama.Framework/src/Metalama.Framework.Engine/Advising/AdviceFactory.cs:1905]: Can be ignored. We never trim spaces and never take them into account.

File: Metalama.Framework/src/Metalama.Framework.Engine/Advising/AdviceFactory.cs [hunk 2]
```diff
@@ -1898,6 +1898,12 @@ public IAdviser<INamespace> WithNamespace(
...
+        {
+            return this.WithDeclaration( targetNamespace );
+        }
+
         using ( this.WithNonUserCode() )
         {
```
[the remark this answers is not in the document]
Comment (gfraiteur) [reply Metalama.Framework/src/Metalama.Framework.Engine/Advising/AdviceFactory.cs:1911]: Actually,   `using ( this.WithNonUserCode() )` should be added to `WithDeclaration`.

**Review by gfraiteur — CHANGES_REQUESTED** [review]
Address copilot's feedback.

File: Metalama.Framework/src/Metalama.Framework.Engine/Advising/AdviceFactory.cs (hunk 1 above)
[the remark this answers is not in the document]
Comment (PostSharpAgent) [reply Metalama.Framework/src/Metalama.Framework.Engine/Advising/AdviceFactory.cs:1905]: Acknowledged. Keeping `IsNullOrEmpty` as-is per @gfraiteur's instruction — whitespace-only names are not trimmed anywhere in the codebase.

— Claude for @gfraiteur

File: Metalama.Framework/src/Metalama.Framework.Engine/Advising/AdviceFactory.cs (hunk 2 above)
[the remark this answers is not in the document]
Comment (PostSharpAgent) [reply Metalama.Framework/src/Metalama.Framework.Engine/Advising/AdviceFactory.cs:1911]: Done. Added `using (this.WithNonUserCode())` to the `WithDeclaration` method body, as @gfraiteur suggested. This ensures all callers of `WithDeclaration` consistently enter the non-user-code scope.

— Claude for @gfraiteur

File: Metalama.Framework/src/Metalama.Framework.Engine/Advising/AdviceFactory.cs [hunk 3]
```diff
@@ -1898,6 +1898,12 @@ public IAdviser<INamespace> WithNamespace(
             throw new InvalidOperationException();
         }
 
+        // An empty name resolves to the target namespace itself.
+        if ( string.IsNullOrEmpty( name ) )
+        {
+            return this.WithDeclaration( targetNamespace );
+        }
```
[the remark this answers is not in the document]
Comment (PostSharpAgent) [reply Metalama.Framework/src/Metalama.Framework.Engine/Advising/AdviceFactory.cs:1908]: The `name` parameter is non-nullable (`string`, not `string?`), so passing `null` is not a supported API usage and would produce a compiler warning. The `IsNullOrEmpty` check is a defensive measure, but adding a test that requires `null!` to bypass nullability would be testing an unsupported scenario. Skipping this one.

— Claude for @gfraiteur

**Review by gfraiteur — APPROVED** [review]
